app/src/main/java/com/babylon/wallet/android/data/repository/state/AccountsStateCache.kt

```diff
@@ -333,6 +356,20 @@ class AccountsStateCache @Inject constructor(
         }
     }
 
+    private fun Flow<MutableMap<AccountAddress, AccountCachedData>>.filterHiddenAssets() = map { cached ->
+        val assetPreferences = profileRepository.profile.first().appPreferences.assets
+        val hiddenPoolAddresses = assetPreferences.hiddenPools().toSet()
+        val hiddenFungibleAddresses = assetPreferences.hiddenFungibles().toSet()
+
+        cached.mapValues { entry ->
+            entry.value.copy(
+                fungibles = entry.value.fungibles.filterNot { it.address in hiddenFungibleAddresses }
```

Comment:
maybe it would be better to check if address is hidden fungible/pool in one filter pass to avoid looping the collection multiple times?

Reply:
Looping at least 2 times is necessary because fungibles and nonFungibles are 2 separate lists. We could filter both the pools and the rest of fungibles in a single loop but I think we lose readability because we'd have 2 different approaches in a single place for the same operation. I would have preferred looping 1 single time if it was possible but since it's not, maybe it's better to keep it as is for consistency.